snuba/query/validation/signature.py

```diff
@@ -104,6 +113,35 @@ def validate(self, expression: Expression, schema: ColumnSet) -> None:
             )
 
 
+class Literal(ParamType):
+    """
+    Validates that the type of a Literal expression is in a set of
+    allowed types.
+
+    If the expression provided is not a Literal, it accepts it.
+    We may consider later whether we want to enforce only literal
+    expressions can be passed as arguments in certain functions.
+    """
+
+    def __init__(self, types: Set[OptionalScalarType], allow_nullable=True) -> None:
```

Comment:
Do you actually need the allow_nullable field? OptionalScalarType already include None.

Reply:
The typing of `None` is a little weird, originally I had intended you to be able to pass `None` in the set of types, e.g. `Literal({None, float})`. However when you do that, this check fails `isinstance(value, tuple(self.__valid_types))` because `None` is not a valid type.

So there was an option with making people pass `type(None)` in the Literal constructor, or add this field making it more explicit about what the validator was doing.

Reply:
OK I addressed this by doing an explicit check for None. You were right, mypy did not like the `type(None)` solution and `isinstance` was not happy with `None` either so I removed them both.